### ISSUE TITLE
Synonym doesn't work as expected

### ISSUE THREAD

**Issue by michielmulders** [meilisearch/MeiliSearch#1135]
**Describe the bug**
Want to create a one-way synonym for `lotr` to `The Lord of The Rings`. I've learned that there's a limitation for the word-length of synonyms, so I've tried to make one for `JP` -> `Jude Fisher` which is an author in my [dataset](https://gist.githubusercontent.com/michielmulders/0f04916b50c826217b3756a682a09d28/raw/15d2077fd685fe522d19e341da636bdde173fcc2/dataset.json).

**To Reproduce**
1. Create `books` index
2. Upload my dataset
3. Create the above two synonyms. My settings object looks like this;

```js
{ rankingRules:
   [ 'typo',
     'words',
     'proximity',
     'attribute',
     'wordsPosition',
     'exactness' ],
  distinctAttribute: null,
  searchableAttributes: [ '*' ],
  displayedAttributes: [ '*' ],
  stopWords: [],
  synonyms: { JF: [ 'Jude Fisher' ], lotr: [ 'The Lord of the Rings' ] },
  attributesForFaceting: [ 'language' ] }
```

**Expected behavior**
I want to find a couple of `The Lord of the Rings` results that are included in my dataset when searching for `lotr` or `JF`.

Currently, I get an empty result `[]`.

**Desktop (please complete the following information):**
 - OS: Windows 10
- Node: v10.18.1
- Using MeiliSearch sandbox

**Comment by MarinPostma** [comment]
hello this is the same instance as #763, #1148 and #964.
 This is due to synonyms being case sensitive, and can be solved by putting. The recent efforts on tokenization now make it possible to normalize synonyms. This issue is prioritized and should be part of the next release.

In the mean time, I suggest that you lowercase all of your synonyms

**Comment by LegendreM** [comment]
Hello @michielmulders,
we plan to fix this bug in 0.19,

thanks for your report!
